platform-sdk/swirlds-common/src/main/java/com/swirlds/common/context/internal/PlatformUncaughtExceptionHandler.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.context.internal;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Simple uncaught exception handler that logs the exception and rethrows it.
+ */
+public class PlatformUncaughtExceptionHandler implements Thread.UncaughtExceptionHandler {
+
+    private static final Logger logger = LogManager.getLogger(PlatformUncaughtExceptionHandler.class);
+
+    @Override
+    public void uncaughtException(Thread t, Throwable e) {
+        logger.error("Uncaught exception in thread: " + t.getName(), e);
```

Comment:
Is the lack of marker intentional? As is, this will not show up in `swirlds.log`.

```suggestion
        logger.error("Uncaught exception in thread: " + t.getName(), e);
```

Reply:
I have no real idea about the logging config today. How should it look like?

Reply:
Discussed offline. Will be handled as a follow up task (possibly by hashgraph team).